docs/topics/virtual-warping.rst

```diff
@@ -0,0 +1,53 @@
+Virtual Warping
+===============
+
+Rasterio has a class that abstracts many of the details of raster warping by
+using an in-memory `Warped VRT
+<http://www.gdal.org/gdal_vrttut.html#gdal_vrttut_warped>`__.
+
+To virtually warp the Rasterio test dataset to EPSG:3857 and extract pixels
+corresponding to its central zoom 9 tile, do the following.
+
+.. code-block:: python
+
+  import mercantile
+
+  from rasterio.io import VirtualWarpedFile
+  from rasterio.enums import Resampling
+
+
+  with VirtualWarpedFile('tests/data/RGB.byte.tif', dst_crs='EPSG:3857',
+                         resampling=Resampling.bilinear).open() as src:
```

Comment:
(This was the beginning of more feedback after using this for something, but I realized quickly that I wasn't using it properly.)